### PR TITLE
Fix for jenkins mirror redirect - #398

### PR DESCRIPTION
**Pull request by samdunne**
Related to #398

### CODE REVIEW

File: libraries/plugin.rb [hunk 1]
```diff
@@ -299,6 +299,8 @@ def install_plugin_from_update_center(plugin_name, plugin_version, opts = {})
     # @option opts [Boolean] :install_deps indicates a plugins dependencies should be installed
     #
     def install_plugin_from_url(source_url, plugin_name, plugin_version = nil, opts = {})
+      source_url = source_url.gsub('http://updates.jenkins-ci.org/download', node['jenkins']['master']['mirror'])
```
Comment (schisamo) [inline libraries/plugin.rb]: Could we add a few lines of comment explaining why this change is necessary?

**Comment by schisamo** [comment]
Let's add some integration test coverage also.

**Comment by samdunne** [comment]
No problem. Thinking further on this it will break existing functionality. I'll add some logic around this so we don't do that.

**Comment by schisamo** [comment]
@samdunne Are you still interested in getting this contribution merged? Waiting on your response to my comments above.

**Comment by samdunne** [comment]
Hey @schisamo sorry I completely forgot about this. I am interested. I'll do some work this week

**Comment by samdunne** [comment]
So the reason for this change is if someone specifies a specific mirror to download from, you would assume that all dependency plugins are also installed from this mirror.

What sort of integration tests are we looking for here?

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.chef.io/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## Non-GitHub Verified Committers

There are 1 commit author(s) whose commits are authored by a non-GitHub verified email address. Chef will have to manually verify that they are authorized to contribute.

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Comment by chef-supermarket** [comment]
Hi. Your friendly Curry bot here. Just letting you know that all commit authors have become authorized to contribute. I have added the "Signed CLA" label to this issue so it can easily be found in the future.

**Comment by schisamo** [comment]
@samdunne Be sure to rebase this branch onto master...I cleaned up various things in #422.

**Comment by schisamo** [comment]
One more rebase :wink: . This just merged change should finally stabilize Travis CI's execution of the integration tests: 
https://github.com/chef-cookbooks/jenkins/commit/1625034b6bb94738e2f7be24527ec15cf1221ca1

**Comment by samdunne** [comment]
Done and done

**Comment by schisamo** [comment]
@samdunne One more things before I merge...what's up with these file renames in your commit?

```
spec/recipes/java_spec.rb → spec/unit/java_spec.rb
spec/recipes/master_spec.rb → spec/unit/master_spec.rb
```

**Comment by samdunne** [comment]
It was to stop foodcritic thinking that ```spec/``` was a cookbook. I can revert if you like.

**Comment by samdunne** [comment]
Anything else needed for this @schisamo?

**Comment by samdunne** [comment]
Any update?

**Comment by iennae** [comment]
@samdunne,

Apologies for the delay in responding here. Can you change it to spec/unit/recipes/*_spec.rb versus spec/unit/

If you could rebase one more time, I'd like to get this added in as it seems like a fairly useful change. Please make sure to sign your commit with git commit -s option to comply with the changed model from CLA to DCO. 

Thanks so much.

**Comment by espoelstra** [comment]
Did anybody ever get this or a similar fix merged? We are using Artifactory as our source of truth/cache for application packages (ie jenkins.war) and our Chef cookbooks/gems as well as hopefully for Jenkins plugins, but the current behavior of grabbing the update-center.json and using those URLs rather than the `jenkins.master.mirror` that we have defined is breaking this pretty definitively.
